### ISSUE TITLE
[bug] Incorrect useState warning on custom hooks named useXState

### ISSUE THREAD

**Issue by lukpsaxo** [Rel1cx/eslint-react#1600]
### Describe the bug

We have some hooks like usePrevState, useAccountState and we are getting errors that useState should be destructured into a getter and setter pair.

### Reproduction

```
function X() {
    const prevBreakpointName = usePrevState(breakpointName);
    return <span>{prevBreakpointName}</span>;
}
```

### Expected behavior

no error

### Platform and versions

```Text
error not in 3.0.0-beta.60 but is in 3.0.0-next.80
```

### Stack trace

```Text

```

**Comment by lukpsaxo** [comment]
from this commit: https://github.com/Rel1cx/eslint-react/commit/8b9a3c337d7bdffc33f7d9daba4193b7eb2d5d7d

seems intentional, but if so, would be good to be able to configure it - I wouldn't want to stop people making useXHooks, I wouldn't expect the need of many useXHooks that return a getter/setter pair.

**Comment by Rel1cx** [comment]
> from this commit: [8b9a3c3](https://github.com/Rel1cx/eslint-react/commit/8b9a3c337d7bdffc33f7d9daba4193b7eb2d5d7d)
> 
> seems intentional, but if so, would be good to be able to configure it - I wouldn't want to stop people making useXHooks, I wouldn't expect the need of many useXHooks that return a getter/setter pair.

That's fair. In the latest beta version, this behaviour has been changed so that extra "useState" patterns are only detected if they've been configured by user via [`additionalStateHooks`](https://www.eslint-react.xyz/docs/configuration/configure-analyzer#additionalstatehooks).
